js/script.js

```diff
@@ -0,0 +1,98 @@
+const formInputs = document.querySelectorAll(".form__input-box");
+const formEmail = document.querySelector("#form__email");
+const formPassword = document.querySelector("#form__password");
+const formSubmit = document.querySelector(".form__submit");
+
+// error 종류에 따라 메시지 추가
+const errorMsgs = {
+  emptyInput: {
+    email: "이메일을 입력해주세요.",
+    password: "비밀번호를 입력해주세요.",
+    text: "비밀번호를 입력해주세요.",
+  },
+  invalidInput: {
+    email: "올바른 이메일 주소가 아닙니다.",
+  },
+  invalidLogin: {
+    email: "이메일을 확인해주세요.",
+    password: "비밀번호를 확인해주세요.",
+  },
+};
+
+function printErrorMsg(error, type, target) {
+  removeErrorMsg(target);
+  const errorMsg = document.createElement("p");
+  errorMsg.textContent = errorMsgs[error][type];
+  errorMsg.classList.add("error-msg");
+  target.after(errorMsg);
+}
+
+function removeErrorMsg(target) {
+  if (
+    target.nextElementSibling &&
+    target.nextElementSibling.classList.contains("error-msg")
+  ) {
+    target.nextElementSibling.remove();
+  }
+}
+
+// input에 값이 존재하는지 확인
+function checkEmptyInput(event) {
+  if (!event.target.value) {
+    event.target.classList.add("form__input-box__error");
+    printErrorMsg("emptyInput", event.target.type, event.target);
+  } else {
+    event.target.classList.remove("form__input-box__error");
+    removeErrorMsg(event.target);
+  }
+}
+
+// input 양식에 맞게 값을 입력했는지 확인
+function validateInput(event) {
+  if (event.target.type == "email" && event.target.value) {
```

Comment:
if 문이 여러번 중첩되는 것은 최대한 지양하는게 좋습니다. 저라면 반대로 type 이 email 이 아니면서 값도 없는 케이스를 체크해서 early return 을 함으로써 중첩을 줄여볼 것 같아요.
추가로 동등연산자 (==) 보다는 일치연산자 (===) 를 사용해보세요!

```suggestion
  if (event.target.type !== "email" && !event.target.value) {
    return;
  }
  
  // ...
```

Reply:
한가지 추가 의견은 현재 validateInput 함수는 오히려 하나의 input 에서만 호출되면 되는 함수인데 input eventListener 코드의 중복을 줄이다보니 오히려 password, password check 하는 input 에서도 이 함수를 호출하는 것 같아요.

type 이 email 인지 체크하는 로직을 함수 호출부에서 체크해서 실제로 필요할 때만 호출될 수 있게 하면 어떨까요?
코드 중복 제거, 재사용도 중요하지만 굳이 필요하지 않은 케이스에서 호출되지 않게 하는 것도 중요한 부분입니다! 👍